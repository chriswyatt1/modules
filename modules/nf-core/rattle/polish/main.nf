process RATTLE_POLISH {
    tag "$meta.id"
    label 'process_low'

    if (params.enable_conda) {
        error "Conda environments are not set up for rattle (when this module was built). Please use docker or singularity containers."
    }
    container 'ecoflowucl/rattle:v1.0'

    input:
    tuple val(meta), path(reads)

    output:
    tuple val(meta), path("${meta.id}.transcriptome.fq"), emit: results
    path "versions.yml"                         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def RATTLE_VERSION = "v1.0" // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    """
    rattle \\
        polish \\
        $args \\
        -t $task.cpus \\
        -i $reads

    mv transcriptome.fq ${meta.id}.transcriptome.fq

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        rattle: $RATTLE_VERSION
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def RATTLE_VERSION = "v1.0" // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    """
    touch ${meta.id}.transcriptome.fq

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        rattle: $RATTLE_VERSION
    END_VERSIONS
    """
}
