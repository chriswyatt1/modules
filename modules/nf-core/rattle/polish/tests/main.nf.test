nextflow_process {

    name "Test Process RATTLE_POLISH"
    script "../main.nf"
    process "RATTLE_POLISH"

    tag "modules"
    tag "modules_nfcore"
    tag "rattle"
    tag "rattle/polish"

    test("test-rattle") {

        when {
            process {
                """
                input[0] = [
				    [ id:'test', single_end:false ], // meta map
				    [ file('${projectDir}/consensi.fq', checkIfExists: true) ]
				]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("test-rattle-stub") {
        options '-stub'
        when {
            process {
                """
                input[0] = [
				    [ id:'test', single_end:false ], // meta map
				    [ file('${projectDir}/consensi.fq', checkIfExists: true) ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
