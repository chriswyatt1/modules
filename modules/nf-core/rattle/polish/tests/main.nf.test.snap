{
    "test-rattle": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "results": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-06T17:34:11.385544474"
    }
}